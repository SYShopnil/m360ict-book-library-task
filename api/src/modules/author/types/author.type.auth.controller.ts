export interface IAuthorLoginBOdy {
  password: string;
  email: string;
}
export interface IPayloadDataOfAccessToken {
  id: number;
  email: string;
}
