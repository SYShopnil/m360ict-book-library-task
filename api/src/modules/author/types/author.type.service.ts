import { IAuthor } from '../../../type/entity';

export interface IGetAllAuthors {
  authors: IAuthor[];
  total: number;
  totalPages: number;
}

export interface IRGetAuthorById extends IAuthor {
  email: string;
}
export interface IRGetAuthorByEmail extends IAuthor {
  email: string;
  password: string;
}
