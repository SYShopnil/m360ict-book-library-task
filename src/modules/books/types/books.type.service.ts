import { IBook } from '../../../type/entity';

export interface GetBooksOptions {
  page: number;
  pageSize: number;
  title?: string;
  author?: string;
}

export interface IRGetBookById extends IBook {}
export interface IRGetBooksByAuthor extends IBook {
  author_name: string;
  authors_bio: string;
}
