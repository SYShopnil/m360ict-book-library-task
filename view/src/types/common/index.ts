export interface ICommonReturnData {
  message: string;
  status: number;
}

export enum EAuth {
  AuthTokenCookieName = "auth",
}

export enum EDataTestId {
  cLoginFormWithSubmit = "cLoginFormWithSubmit",
  cLogoutContainer = "cLogoutContainer",
  rUserList = "rUserList",
  SHeaderMain = "SHeaderMain",
  SProductSection = "SProductSection",
  SProfile = "SProfile",
  CPaginationTrack = "CPaginationTrack",
  RSingleProduct = "RSingleProduct",
  RTableBody = "RTableBody",
  SProductCard = "SProductCard",
  CProductSearchBarContainer = "CProductSearchBarContainer",
}
