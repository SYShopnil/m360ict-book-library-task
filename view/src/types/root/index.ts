export type { IIconStore } from "./_icon";
export { IconName as IconEnums } from "./_icon";
export type { IButton, INormalButton, IRedirectButton } from "./Button";
export { BtnColorSchema } from "./Button";
