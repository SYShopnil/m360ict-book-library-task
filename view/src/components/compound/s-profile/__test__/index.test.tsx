import { render, screen } from "@testing-library/react";
import { SProfile } from "..";
import { EDataTestId } from "@src/types/common";

describe("When Profile has rendered", () => {
  it("Expect render successfully", async () => {
    // const { getByTestId } = render(await SProfile());
    // const myElement = getByTestId(EDataTestId.SProfile);
    const myElement = screen.getByTestId(EDataTestId.SProfile);
    expect(myElement).toMatchSnapshot();
  });
});
