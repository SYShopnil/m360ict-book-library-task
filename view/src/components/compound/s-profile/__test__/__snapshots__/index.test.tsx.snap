// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Profile has rendered Expect render successfully 1`] = `
<div
  data-testid="SProfile"
>
  No User Found!!!
</div>
`;
